src/datatype/svecf32.rs

```diff
@@ -0,0 +1,770 @@
+        let layout1 = Layout::array::<u16>(len).unwrap();
+        let layout2 = Layout::array::<F32>(len).unwrap();
+        let layout = layout.extend(layout1).unwrap().0.pad_to_align();
+        layout.extend(layout2).unwrap().0.pad_to_align()
+    }
+    pub fn new_in_postgres(vector: SparseF32Ref<'_>) -> SVecf32Output {
+        unsafe {
+            let layout = SVecf32::layout(vector.length() as usize);
+            let ptr = pgrx::pg_sys::palloc(layout.size()) as *mut SVecf32;
+            ptr.cast::<u8>().add(layout.size() - 8).write_bytes(0, 8);
+            std::ptr::addr_of_mut!((*ptr).varlena).write(SVecf32::varlena(layout.size()));
+            std::ptr::addr_of_mut!((*ptr).dims).write(vector.dims);
+            std::ptr::addr_of_mut!((*ptr).kind).write(2);
+            std::ptr::addr_of_mut!((*ptr).reserved).write(0);
+            std::ptr::addr_of_mut!((*ptr).len).write(vector.length());
+            std::ptr::addr_of_mut!((*ptr).padding).write_bytes(0, 6);
+            let mut data_ptr = (*ptr).phantom.as_mut_ptr().cast::<u16>();
+            std::ptr::copy_nonoverlapping(
+                vector.indexes.as_ptr(),
+                data_ptr,
+                vector.length() as usize,
+            );
+            data_ptr = data_ptr.add(vector.length() as usize);
+            let offset = data_ptr.align_offset(8);
+            std::ptr::write_bytes(data_ptr, 0, offset);
+            data_ptr = data_ptr.add(offset);
+            std::ptr::copy_nonoverlapping(
+                vector.values.as_ptr(),
+                data_ptr.cast(),
+                vector.length() as usize,
+            );
+            SVecf32Output(NonNull::new(ptr).unwrap())
+        }
+    }
+    pub fn new_zeroed_in_postgres(len: usize) -> SVecf32Output {
+        unsafe {
+            let layout = SVecf32::layout(len);
+            let ptr = pgrx::pg_sys::palloc0(layout.size()) as *mut SVecf32;
+            ptr.cast::<u8>().add(layout.size() - 8).write_bytes(0, 8);
+            std::ptr::addr_of_mut!((*ptr).varlena).write(SVecf32::varlena(layout.size()));
+            std::ptr::addr_of_mut!((*ptr).kind).write(2);
+            std::ptr::addr_of_mut!((*ptr).reserved).write(0);
+            std::ptr::addr_of_mut!((*ptr).len).write(len as u16);
+            SVecf32Output(NonNull::new(ptr).unwrap())
+        }
+    }
+    pub fn dims(&self) -> u16 {
+        self.dims
+    }
+    pub fn len(&self) -> usize {
+        self.len as usize
+    }
+    fn indexes(&self) -> *const u16 {
+        self.phantom.as_ptr().cast()
+    }
+    fn values(&self) -> *const F32 {
+        let len = self.len as usize;
+        unsafe {
+            let ptr = self.phantom.as_ptr().cast::<u16>().add(len);
+            let offset = ptr.align_offset(8);
+            ptr.add(offset).cast()
+        }
+    }
+    fn indexes_mut(&mut self) -> *mut u16 {
+        self.phantom.as_mut_ptr().cast()
+    }
+    fn values_mut(&mut self) -> *mut F32 {
```

Comment:
Remove it.

Reply:
`_vectors_svecf32_recv` needs it. I updated the returned value from a pointer to a reference.